library/alignment.h

```diff
@@ -181,6 +181,7 @@ inline void mbedtls_put_unaligned_uint64(void *p, uint64_t x)
 
 /* Detect armcc built-in byteswap routine */
 #if defined(__ARMCC_VERSION) && (__ARMCC_VERSION >= 410000) && !defined(MBEDTLS_BSWAP32)
```

Comment:
According to https://developer.arm.com/documentation/101028/0004 we should also check `defined(__ARM_ACLE)`, and possibly use `__ARM_ACLE >= 201` (or maybe some smaller number?) rather than check `__ARMCC_VERSION`.

Reply:
Actually, the relevant documentation for ARM Compiler 5 is [this](https://developer.arm.com/documentation/dui0472/m/Compiler-specific-Features/--rev-intrinsic?lang=en), and it doesn't mention a header. ARM Compiler 5 does not ship with a file called `arm_acle.h` and it doesn't seem to be built into the binary either. So we need something like this:

```
#if defined(__ARM_ACLE) // ARM Compiler 6 (earlier versions don't need a header)
#include <arm_acle.h>
#endif
#define MBEDTLS_BSWAP32 __rev
```

Reply:
Trying to reverse engineer the reason it checks for 4.1, I think this is the oldest version I could find documentation for. Maybe 4.0 also supports __rev but I couldn't find a definitive answer. So I suggest keeping this version check and adding the `__ARM_ACLE` check as Gilles suggests.